source/main/ipc/getHardwareWalletChannel.ts

```diff
@@ -868,56 +874,13 @@ export const handleHardwareWalletRequests = async (
       throw e;
     }
   });
-  // @ts-ignore ts-migrate(2345) FIXME: Argument of type '(params: TrezorSignTransactionRe... Remove this comment to see the full error message
-  signTransactionTrezorChannel.onRequest(async (params) => {
-    const {
-      inputs,
-      outputs,
-      protocolMagic,
-      fee,
-      ttl,
-      networkId,
-      certificates,
-      devicePath,
-      validityIntervalStartStr,
-      withdrawals,
-      auxiliaryData,
-      signingMode,
-    } = params;
-
-    if (!TrezorConnect) {
-      throw new Error('Device not connected!');
-    }
-
-    try {
-      const dataToSign = {
-        inputs,
-        outputs,
-        fee,
-        ttl,
-        protocolMagic,
-        networkId,
-        certificates,
-        withdrawals,
-        validityIntervalStartStr,
-        auxiliaryData,
-        signingMode,
-      };
-
-      logger.info(
-        '[TREZOR-CONNECT] Calling TrezorConnect.cardanoSignTransaction()'
-      );
 
-      // @ts-ignore ts-migrate(2345) FIXME: Argument of type '{ inputs: TrezorSignTransactionI... Remove this comment to see the full error message
-      const signedTransaction = await TrezorConnect.cardanoSignTransaction({
-        device: buildTrezorDeviceParams(devicePath),
-        ...dataToSign,
-      });
+  signTransactionTrezorChannel.onRequest((dataToSign) => {
+    logger.info(
+      '[TREZOR-CONNECT] Calling TrezorConnect.cardanoSignTransaction()'
+    );
 
-      return Promise.resolve(signedTransaction);
-    } catch (e) {
```

Comment:
Again, this try-catch block was reduntant, if error is immediately rethrown without any additional action, it is the same as not wrapping the code in try-catch block at all.